plugins/inputs/consul/consul.go

```diff
@@ -39,9 +40,21 @@ func (*Consul) SampleConfig() string {
 
 func (c *Consul) Init() error {
 	if c.MetricVersion != 2 {
-		c.Log.Warnf("Use of deprecated configuration: 'metric_version = 1'; please update to 'metric_version = 2'")
+		telegrafConfig.PrintOptionValueDeprecationNotice("inputs.consul", "metric_version", 1,
+			telegraf.DeprecationInfo{
+				Since:     "1.16.0",
+				RemovalIn: "1.40.0",
+				Notice:    `please update to 'metric_version = 2'`,
+			},
+		)
 	}
 
+	newClient, err := c.createAPIClient()
+	if err != nil {
+		return err
+	}
+	c.client = newClient
```

Comment:
Are you 100% sure this does not attempt to create a connection? Otherwise this change is a change in behavior of the plugin. Anyway, the complete change to this file seems unrelated to the PR title. I would vote for a separate PR if you whish to pursue those changes.

Reply:
Yup, see https://github.com/hashicorp/consul/blob/api/v1.29.4/api/api.go#L674

Reply:
Anyway, please move this to an own PR so we can revert it independently if we run into trouble.

Reply:
Okay, reverted in c29243cd4dd05f96daf3986b5f9f5c783eac1c31.